Tribler/Test/Core/Libtorrent/test_libtorrent_download_impl.py

```diff
@@ -140,3 +129,47 @@ def test_multifile_torrent(self):
         test_dict["a"] = "b"
         pstate.set("state", "engineresumedata", test_dict)
         return impl.network_create_engine_wrapper(pstate)
+
+    @deferred(timeout=10)
+    def test_save_resume(self):
+        """
+        testing call resume data alert
+        """
+        tdef = self.create_tdef()
+
+        impl = LibtorrentDownloadImpl(self.session, tdef)
+
+        def resume_ready(d=None):
+            """
+            check if resume data is ready
+            """
+            if not d:
+                d = defer.Deferred()
+
+            if not impl.pstate_for_restart:
+                reactor.callLater(2, resume_ready, d)
```

Comment:
@whirm, I think it's ready. I put those deferreds in an array to make sure all call is not overwritten by another call